matchms/filtering/SpectrumProcessor.py

```diff
@@ -318,7 +318,9 @@ def to_dataframe(self):
             if filter_name not in processing_report["filter"].values:
                 processing_report.loc[len(processing_report)] = {"filter": filter_name}
 
-        processing_report = processing_report.set_index("filter").fillna(0)
+        with pd.option_context("future.no_silent_downcasting", True):
```

Comment:
This seems to only work for Pandas >= 2.2
I got errors for Pandas 2.0.* and 2.1.*

Reply:
Maybe add a try-except or check for the present pandas version?